util/abs/provision_pe_xl_nodes.rb

```diff
@@ -12,40 +86,39 @@
 ROLES_HA = %w[master_replica
               puppet_db_replica].freeze
 
-# TODO: update to use OptionParser
-# NOTE: set HA to false for a non-HA environment
-HA = false
-ROLES = if HA
-          ROLES_CORE + ROLES_HA
-        else
-          ROLES_CORE
-        end
+if options[:ha]
+  HA = true
+  ROLES = ROLES_CORE + ROLES_HA
+else
+  HA = false
+  ROLES = ROLES_CORE
+end
 
-# TODO: update to use OptionParser
-# currently uses ARGV[0], ARGV[1] if specified, otherwise these defaults
-ABS_ID = "slv"
-OUTPUT_DIR = "./"
+NOOP = options[:noop] || false
```

Comment:
When I write defaults, I put them in constants, so that I can refer to them in the help text as well as use them in this section... then have one place to change them if I need to.

Reply:
Yes, these constants were originally the defaults. I tried to add the options with as little impact to the code as possible; since the constants were already being used in the code it seemed to make sense to leverage them rather than updating to use new variables. I have added the defaults as constants before the description (since they are now used in it).